File: Day243/getFirstSetBit.java
package Day243;

public class getFirstSetBit {

    public static int findFirstSetBit(int n) {
        if (n == 0) {
            return 0;
        }
        int answer = 1;
        while (n!=0) {
            if ((n&1)==0) {
                answer++;
                n >>= 1;
            } else {
                return answer;
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        int n = 12;
        System.out.println(findFirstSetBit(n));
    }
}

File: Day243/getFirstSetBit.go
package main 

import (
	"fmt"
)

func findFirstSetBit(n int) int {
	if n == 0 {
		return 0
	} 
	answer := 1
	for n != 0 {
		if (n&1) == 0 {
			answer++
			n >>= 1
		} else {
			return answer
		}

	}
	return answer
}

func main() {
	fmt.Println(findFirstSetBit(8))
	
}